airflow/providers/amazon/aws/sensors/sqs.py

```diff
@@ -131,19 +165,7 @@ def poll_sqs(self, sqs_conn: BaseAwsConnection) -> Collection:
             receive_message_kwargs["VisibilityTimeout"] = self.visibility_timeout
 
         response = sqs_conn.receive_message(**receive_message_kwargs)
-
-        if "Messages" not in response:
-            return []
-
-        messages = response["Messages"]
-        num_messages = len(messages)
-        self.log.info("Received %d messages", num_messages)
-
-        if num_messages and self.message_filtering:
-            messages = self.filter_messages(messages)
-            num_messages = len(messages)
-            self.log.info("There are %d messages left after filtering", num_messages)
-        return messages
+        return response
```

Comment:
Why do we change the return value from messages to response? Won't it change the behavior?

Reply:
No it doesn't change the behavior because the original returned value, `messages`, is still being used in the `poke` method. All I'm doing is pulling out the logic of parsing the response (from lines 135 to 146 in the original code) to a separate function so it can be reused in the Trigger. The method `process_response` takes the response from `poll_sqs` and returns `messages`, which then gets used, just like it was before.

Reply:
Got it.  Sounds good 👍